libcaf_io/src/default_multiplexer.cpp

```diff
@@ -279,19 +280,159 @@ namespace network {
+                       pollset_.data(), static_cast<int>(pollset_.size()),
+                       nullptr);
+      CAF_LOG_DEBUG("kevent() on " << CAF_ARG(pollset_.size())
+                    << " sockets reported " << CAF_ARG(nev)
+                    << " event(s)");
+      if (nev < 0)
+        switch (errno) {
+          case EINTR:
+            shadow_.changes.clear(); // all changes guaranteed to be applied
+            continue;
+          default:
+            CAF_LOG_ERROR("kevent: " << strerror(errno));
+        }
+      shadow_.changes.clear();
+      auto iter = pollset_.begin();
+      auto last = iter + nev;
+      for (; iter != last; ++iter) {
+        auto fd = static_cast<native_socket>(iter->ident);
+        auto ptr = reinterpret_cast<event_handler*>(iter->udata);
+        CAF_ASSERT(ptr != nullptr);
+        if (iter->flags & EV_ERROR || iter->flags & EV_EOF) {
+          CAF_LOG_DEBUG("error occured on socket:"
+                        << CAF_ARG(fd) << CAF_ARG(iter->data)
+                        << CAF_ARG(strerror(iter->data)));
+          auto i = std::lower_bound(shadow_.fds.begin(), shadow_.fds.end(), fd);
+          if (i != shadow_.fds.end() && *i == fd) {
+            ptr->handle_event(operation::propagate_error);
+            shadow_.fds.erase(i);
+          }
+          // Pending events for the failed handler are no longer relevant.
+          auto e = std::lower_bound(events_.begin(), events_.end(), fd,
+                                    event_less{});
+          while (e != events_.end() && e->fd == fd)
+            e = events_.erase(e);
+        } else if (iter->filter == EVFILT_READ && !ptr->read_channel_closed()) {
+          ptr->handle_event(operation::read);
+        } else if (iter->filter == EVFILT_WRITE) {
+          ptr->handle_event(operation::write);
+        } else {
+          CAF_ASSERT(!"unexpected kevent filter");
+        }
+      }
+      // Handle accumulated events.
+      for (auto& me : events_)
+        handle(me);
+      events_.clear();
+    }
+  }
+
+  void default_multiplexer::handle(const default_multiplexer::event& e) {
+    CAF_LOG_TRACE("e.fd = " << CAF_ARG(e.fd) << ", mask = "
+                  << CAF_ARG(e.mask));
+    // ptr is only allowed to nullptr if fd is our pipe
+    // read handle which is only registered for input
+    CAF_ASSERT(e.ptr != nullptr || e.fd == pipe_.first);
+    if (e.ptr != nullptr && e.ptr->eventbf() == e.mask)
+      return;
+    auto old = e.ptr != nullptr ? e.ptr->eventbf() : input_mask;
+    if (e.ptr)
+      e.ptr->eventbf(e.mask);
+    auto kqueue_update = [&](unsigned short flag, short filter) {
+      CAF_ASSERT(flag == EV_ADD || flag == EV_DELETE);
+      CAF_ASSERT(filter == EVFILT_WRITE || filter == EVFILT_READ);
+      struct kevent ke;
+      EV_SET(&ke, e.fd, filter, flag, 0, 0, e.ptr);
+      shadow_.changes.emplace_back(std::move(ke));
+    };
+    if (e.mask == 0) {
+      CAF_LOG_DEBUG("attempt to remove socket" << CAF_ARG(e.fd)
```

Comment:
`"attempt to remove socket e.fd = ... from kqueue"` reads a bit odd in the logfile, don't you think? I'd log `e.fd` without `CAF_ARG` here or put it at the end, e.g., `CAF_LOG_DEBUG("attempt to ... from kqueue:" << CAF_ARG(e.fd))`.

The same applies for the logging in lines 393 and 402.